app/javascript/components/get-weather-query.jsx

```diff
@@ -41,46 +59,61 @@ export const WeatherQuery = ({
   DisconnectedMessage,
   WeatherFrag,
   queryName,
-}) => (
-  <Query query={getPrimaryLocationWeather(WeatherFrag, queryName)}>
-    {({ loading, error, data, subscribeToMore }) => {
-      if (loading) {
-        return <LoadingMessage />;
+  location,
+}) => {
+  const cityName = location.pathname.substring(1);
```

Comment:
react-router passes props to its target component:
https://reactrouter.com/web/api/Route/route-props
https://reactrouter.com/web/api/match

You should be able to retrieve the matched `city_name` from the match prop. This will probably require passing this from WidgetWindow to WidgetController. Doing so would remove the need for the `withRouter` higher order component.

I think passing the `cityName` property directly will keep our queries independent of the url logic.

Reply:
Are you referring to the useRouteMatch hook? I had trouble with that because I couldn't use it unless it nested in a functional component. Are you saying to reference match via props in the same way that location is referenced via props? Would I not need withRouter in that case?

Reply:
Nope. I mean `<Route ....` passes props to `component`, one of them is `match` which is based on the pattern(s) in the path. Try out:
```
const WidgetWindow = (props) => {
  console.log(props); // or you could: debugger;
  return <WidgetController client={client} />;
};
```